database/repository/header_repository.go

```diff
@@ -133,6 +133,41 @@ func NewHeadersRepository(db *sql.HeadersDb) *HeaderRepository {
 	return &HeaderRepository{db: db}
 }
 
+func (r *HeaderRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey int) (*domains.MerkleRootsESKPagedResponse, error) {
+	orderByField := "BlockHeight"
+	sortDirection := "ASC"
+	merklerootsFromDb, err1 := r.db.GetMerkleRoots(batchSize, lastEvaluatedKey)
+	merklerootsTopHeight, err2 := r.GetHeadersCount()
+
+	if err1 != nil {
+		return nil, err1
+	}
+
+	if err2 != nil {
+		return nil, err2
+	}
+
+	merkleroots := &domains.MerkleRootsESKPagedResponse{
+		Page: domains.ExclusiveStartKeyPage[int]{
+			OrderByField:  &orderByField,
+			SortDirection: &sortDirection,
+			TotalElements: int32(merklerootsTopHeight),
+			Size:          len(merklerootsFromDb),
+			// Last Evaluated Key is the height of the last element returned from the merklerootsFromDb
+			LastEvaluatedKey: int(merklerootsFromDb[len(merklerootsFromDb)-1].Height),
+		},
+	}
+
+	for _, merkleroot := range merklerootsFromDb {
+		merkleroots.Content = append(merkleroots.Content, &domains.MerkleRootsResponse{
+			MerkleRoot:  merkleroot.MerkleRoot,
+			BlockHeight: merkleroot.Height,
+		})
+	}
```

Comment:
For efficiency sake, would be good to reserve memory at once - since we know the size of merklerootsFromDb
```golang
merkleroots.Content = make([]*domains.MerkleRootsResponse, 0, len(merklerootsFromDb))
for i, merkleroot := range merklerootsFromDb {
	merkleroots.Content[i] = &domains.MerkleRootsResponse{
		MerkleRoot:  merkleroot.MerkleRoot,
		BlockHeight: merkleroot.Height,
	}
}
```

Reply:
Ah nice catch, applied changes